src/knapsack/SimpleKnapsackModel.py

```diff
@@ -0,0 +1,131 @@
+  """
+
+  @classmethod
+  def getInputSpecs(cls):
+    """
+      Collects input specifications for this class.
+      @ In, None
+      @ Out, inputSpecs, InputData, input specifications
+    """
+    inputSpecs = InputData.parameterInputFactory('ExternalModel')
+    inputSpecs.addParam('name', param_type=InputTypes.StringType, required=True)
+    inputSpecs.addParam('subType', param_type=InputTypes.StringType, required=True)
+
+    inputSpecs.addSub(InputData.parameterInputFactory('capacity', contentType=InputTypes.StringType))
+    inputSpecs.addSub(InputData.parameterInputFactory('penaltyFactor', contentType=InputTypes.FloatType))
+    inputSpecs.addSub(InputData.parameterInputFactory('outcome', contentType=InputTypes.StringType))
+    inputSpecs.addSub(InputData.parameterInputFactory('choiceValue', contentType=InputTypes.StringType))
+    inputSpecs.addSub(InputData.parameterInputFactory('variables', contentType=InputTypes.StringListType))
+
+    map = InputData.parameterInputFactory('map', contentType=InputTypes.StringType)
+    map.addParam('value', param_type=InputTypes.StringType, required=True)
+    map.addParam('cost', param_type=InputTypes.StringType, required=True)
+    inputSpecs.addSub(map)
+
+    return inputSpecs
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    ExternalModelPluginBase.__init__(self)
+
+    self.capacity      = None    # capacity value of the knapsack
+    self.penaltyFactor = 1.0     # penalty factor that is used when the capacity constraint is not satisfied
+    self.outcome       = None    # ID of the variable which indicates if the chosen elements satisfy the capacity constraint
+    self.choiceValue   = None    # ID of the variable which indicates the sum of the values of the chosen project elements
+
+  def _readMoreXML(self, container, xmlNode):
+    """
+      Method to read the portion of the XML that belongs to the BaseKnapsack model
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, xmlNode, xml.etree.ElementTree.Element, XML node that needs to be read
+      @ Out, None
+    """
+    container.mapping    = {}
+
+    specs = self.getInputSpecs()()
+    specs.parseNode(xmlNode)
+    for node in specs.subparts:
+      name = node.getName()
+      val = node.value
+      if name == 'capacity':
+        self.capacity = val
+      elif name == 'penaltyFactor':
+        self.penaltyFactor = val
+      elif name == 'outcome':
+        self.outcome = val
+      elif name == 'choiceValue':
+        self.choiceValue = val
+      elif name == 'map':
+        container.mapping[val] = [node.parameterValues['value'],node.parameterValues['cost']]
+      elif name == 'variables':
+        variables = val
+      else:
+        raise IOError("BaseKnapsackModel: xml node " + name + " is not allowed")
+
+  def initialize(self, container, runInfoDict, inputFiles):
```

Comment:
This one will be removed if you do not need it.

Reply:
yup